pkg/service/model_build_load_balancer.go

```diff
@@ -232,42 +232,94 @@ func (t *defaultModelBuildTask) resolveLoadBalancerSubnets(ctx context.Context,
 }
 
 func (t *defaultModelBuildTask) buildLoadBalancerAttributes(_ context.Context) ([]elbv2model.LoadBalancerAttribute, error) {
-	var attrs []elbv2model.LoadBalancerAttribute
-	accessLogEnabled := t.defaultAccessLogS3Enabled
-	bucketName := t.defaultAccessLogsS3Bucket
-	bucketPrefix := t.defaultAccessLogsS3Prefix
-	if _, err := t.annotationParser.ParseBoolAnnotation(annotations.SvcLBSuffixAccessLogEnabled, &accessLogEnabled, t.service.Annotations); err != nil {
+	defaultAttribs := t.getDefaultAnnotationLbAttributes()
+	configMapAttribs, err := t.getConfigMapLbAttributes()
+	if err != nil {
 		return []elbv2model.LoadBalancerAttribute{}, err
 	}
-	if accessLogEnabled {
-		t.annotationParser.ParseStringAnnotation(annotations.SvcLBSuffixAccessLogS3BucketName, &bucketName, t.service.Annotations)
-		t.annotationParser.ParseStringAnnotation(annotations.SvcLBSuffixAccessLogS3BucketPrefix, &bucketPrefix, t.service.Annotations)
-	}
-	crossZoneEnabled := t.defaultLoadBalancingCrossZoneEnabled
-	if _, err := t.annotationParser.ParseBoolAnnotation(annotations.SvcLBSuffixCrossZoneLoadBalancingEnabled, &crossZoneEnabled, t.service.Annotations); err != nil {
+	specificAttribs, err := t.getAnnotationSpecificLbAttributes()
+	if err != nil {
 		return []elbv2model.LoadBalancerAttribute{}, err
 	}
+	mergedAttribs := mergeAttributeMaps(defaultAttribs, mergeAttributeMaps(configMapAttribs, specificAttribs))
+	return mkAttributesSliceFromConfigMap(mergedAttribs), nil
+}
+
+func mergeAttributeMaps(firstMap map[string]string, secondMap map[string]string) map[string]string {
+	for key, value := range secondMap {
+		firstMap[key] = value
+	}
+	return firstMap
+}
+
+func mkAttributesSliceFromConfigMap(configMapAttrs map[string]string) []elbv2model.LoadBalancerAttribute {
+	attributes := make([]elbv2model.LoadBalancerAttribute, 0, len(configMapAttrs))
+	keys := make([]string, 0, len(configMapAttrs))
+	for k := range configMapAttrs {
+		keys = append(keys, k)
+	}
+	sort.Strings(keys)
```

Comment:
no need to sort here. You could sort and compare in the unit tests instead.

Reply:
It would be really hard to sort in unit tests or at least I don't know how to do it.
The structure of tests that were failing is as follows:
1. There's a raw json string passed as a `wanted` value: https://github.com/kubernetes-sigs/aws-load-balancer-controller/blob/main/pkg/service/model_builder_test.go#L109
2. There's a single validation method that checks if both json strings are equal from the point of json document view (particular keys ordering doesn't matter, list ordering does matter) 
https://github.com/kubernetes-sigs/aws-load-balancer-controller/blob/main/pkg/service/model_builder_test.go#L2100

These tests implicitly depend on the ordering and I've no idea how to sort in the test itself. Could you please elaborate on how would you sort inside the test here?

Reply:
I was referring to the unit test Test_defaultModelBuilderTask_buildLBAttributes where you can sort the output and the expectation before comparing.

You are right on the Test_defaultModelBuilderTask_Build test. I assumed that with the support for the default removed, the model generated in the Test_defaultModelBuilderTask_Build function would not have the load balancer attributes - but I just realized some tests require it.

could use `sort.Slice` instead of sorting the keys

Reply:
Alright, thanks! I addressed the rest of your comments in code, was just waiting for the response on that one. I'll improve sorting and push another commit later in the day and let you know.